node/src/blockchain/blockchains.rs

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) 2019-2021, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::blockchain::blockchain_interface::{
+    MAINNET_CONTRACT_ADDRESS, MULTINODE_TESTNET_CONTRACT_ADDRESS, RINKEBY_TESTNET_CONTRACT_ADDRESS,
+    ROPSTEN_TESTNET_CONTRACT_ADDRESS,
+};
+use crate::blockchain::blockchains_specific_constants::{
+    MAINNET_CONTRACT_CREATION_BLOCK, MULTINODE_TESTNET_CONTRACT_CREATION_BLOCK,
+    RINKEBY_TESTNET_CONTRACT_CREATION_BLOCK, ROPSTEN_TESTNET_CONTRACT_CREATION_BLOCK,
+};
+use crate::sub_lib::neighborhood::Blockchain;
+use masq_lib::constants::{DEV_LABEL, ETH_MAINNET_LABEL, ETH_RINKEBY_LABEL, ETH_ROPSTEN_LABEL};
+use web3::types::Address;
+
+pub const KEY_VS_IP_DELIMITER: char = ':';
+pub const CHAIN_LABEL_DELIMITER: char = '.';
+
+pub const CHAINS: [BlockchainRecord; 4] = [
+    BlockchainRecord {
+        num_chain_id: 1,
+        non_num_chain_id: Blockchain::EthMainnet,
+        in_command_name: "eth-mainnet",
+        directory_by_platform: "eth",
+        chain_label: ETH_MAINNET_LABEL,
+        contract: MAINNET_CONTRACT_ADDRESS,
+        contract_creation_block: MAINNET_CONTRACT_CREATION_BLOCK,
+    },
+    BlockchainRecord {
+        num_chain_id: 2,
+        non_num_chain_id: Blockchain::EthRopsten,
+        in_command_name: "ropsten",
+        directory_by_platform: "eth",
+        chain_label: ETH_ROPSTEN_LABEL,
+        contract: ROPSTEN_TESTNET_CONTRACT_ADDRESS,
+        contract_creation_block: ROPSTEN_TESTNET_CONTRACT_CREATION_BLOCK,
+    },
+    BlockchainRecord {
+        num_chain_id: 3,
+        non_num_chain_id: Blockchain::EthRinkeby,
+        in_command_name: "rinkeby",
+        directory_by_platform: "eth",
+        chain_label: ETH_RINKEBY_LABEL,
+        contract: RINKEBY_TESTNET_CONTRACT_ADDRESS,
+        contract_creation_block: RINKEBY_TESTNET_CONTRACT_CREATION_BLOCK,
+    },
+    BlockchainRecord {
+        num_chain_id: 4,
+        non_num_chain_id: Blockchain::Dev,
+        in_command_name: "dev",
+        directory_by_platform: "dev",
+        chain_label: DEV_LABEL,
+        contract: MULTINODE_TESTNET_CONTRACT_ADDRESS,
+        contract_creation_block: MULTINODE_TESTNET_CONTRACT_CREATION_BLOCK,
+    },
+];
+
+#[derive(Debug, PartialEq)]
+pub struct BlockchainRecord {
+    pub num_chain_id: u8,
+    pub non_num_chain_id: Blockchain,
+    pub in_command_name: &'static str,
+    pub directory_by_platform: &'static str,
+    pub chain_label: &'static str,
+    pub contract: Address,
+    pub contract_creation_block: u64,
+}
+
+#[track_caller]
+fn return_right_record<F>(closure: Box<F>) -> &'static BlockchainRecord
```

Comment:
This method name is awkward. I think this could be implemented using a Map<String, BlockchainRecord> where a simple Map::get() -> Option<BlockchainRecord> (I know this is not valid Rust but hopefully it conveys the idea)